tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Azure.Sdk.Tools.CodeOwnersParser;
+using NUnit.Framework;
+
+namespace Azure.Sdk.Tools.RetrieveCodeOwners.Tests
+{
+    public class MainTests
+    {
+        private const string codeOwnerFilePath = "CODEOWNERS";
+        private List<Tuple<string, bool, List<string>>> args { get; set; } = new List<Tuple<string, bool, List<string>>>()
+        {
+            new Tuple<string, bool, List<string>>("sdk", false, new List<string> { "person1", "person2" }),
+            new Tuple<string, bool, List<string>>("/sdk", false, new List<string> { "person1", "person2" }),
+            new Tuple<string, bool, List<string>>("sdk/noPath", false, new List<string> { "person1", "person2"}),
+            new Tuple<string, bool, List<string>>("/sdk/azconfig", false, new List<string> { "person3", "person4"}),
+            new Tuple<string, bool, List<string>>("/sdk/azconfig/package", false, new List<string> { "person3", "person4"}),
+            new Tuple<string, bool, List<string>>("/sdk/testUser/", true, new List<string> { "azure-sdk" }),
+            new Tuple<string, bool, List<string>>("/sd", true, new List<string>()),
+        };
+
+        [Test]
```

Comment:
Instead of having only a single test you should consider testcase https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcase.html so that we have better reporting of which case is actually failing.

Reply:
I am just wondering if we have something like parameterized tests. Thanks for pointing this out.